sentry.c

```diff
@@ -0,0 +1,124 @@
+#include <cheri/cheric.h>
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+
+#include "include/common.h"
+#include "include/instructions.h"
+#include "include/regs.h"
+
+struct data {
+	int a;
+	int b;
+	char c[256];
+};
+
+void* setup_sentry(void*, void (*f)(uint32_t*, void*));
+void oop_sentry(struct data*, int);
+void simple_sentry(int);
+
+void oop_sentry(struct data* this, int arg) {
+	printf("OOP: %p %d \n", this, arg);
+}
+
+void simple_sentry(int arg) {
+	printf("Simple: %d \n", arg);
+}
+
+void gen_oop(uint32_t* code, void* function) {
+	intptr_t* code_data = (intptr_t*)code;
+	code_data[255] = (intptr_t)function;
+	code_data[254] = (intptr_t)malloc(sizeof(struct data));
+	
+	uint32_t idx = 0;
+	code[idx++] = auipcc(cs2, 1);
+	code[idx++] = clc_128(cs3, cs2, ((-16) + (2 << 20)));
+	code[idx++] = addi(a1, a0, 0);
+	code[idx++] = addi(a2, a1, 0);
+	code[idx++] = addi(a3, a2, 0);
+	code[idx++] = addi(a4, a3, 0);
+	code[idx++] = addi(a5, a4, 0);
+	code[idx++] = addi(a6, a5, 0);
+	code[idx++] = addi(a7, a6, 0);
+	code[idx++] = clc_128(ca0, cs2, ((-32) + (2 << 20)));
+	code[idx++] = cjalr(cnull, cs3);
+
```

Comment:
Extraneous blank lines.

Reply:
It pains me to ask for this, as I prefer `type* name` over `type *name`, but this needs to be passed by `clang-format` as well.

Reply:
I should add a commit hook to format before pushing.